### ISSUE TITLE
Rule suggestion: add "allow-arrow-return-shorthand" for rule `no-void-expression`

### ISSUE THREAD

**Issue by vilic** [palantir/tslint#2473]
Consider this:

```ts
(() => fnThatMightThrow('foo', 'bar'))
    .should.throw();
```
